lib/web/desktop.go

```diff
@@ -392,6 +392,33 @@ func proxyWebsocketConn(ws *websocket.Conn, wds net.Conn) error {
 	return trace.NewAggregate(retErrs...)
 }
 
+// wasWebsocketClosedByClient handles the error returned by proxyWebsocketConn
+func handleProxyWebsocketConnErr(err error, log *logrus.Entry) {
+	if err == nil {
+		return
+	}
+
+	if agg, ok := trace.Unwrap(err).(trace.Aggregate); ok {
```

Comment:
Should we type switch and loop over `errors.Unwrap`, for added resilience to other types of wrapping?

Eg, something like this:

```go
errs = []error{websocketErr}
for len(errs) > 0 {
	err := errs[0] // pop first error
	errs = errs[1:]

	switch err := err.(type) {
	case trace.Aggregate:
		errs = append(errs, err.Errors())
	case *websocket.CloseError:
		// Check codes, log, etc.
		return
	default:
		if wrapped := errors.Unwrap(err); wrapped != nil {
			errs = append(errs, wrapped)
		}
	}
}
```

Reply:
d9d1d6f87852a508b34da1da7665c013ff785feb